src/src/rx-serial/devSerialIO.cpp

```diff
@@ -29,20 +36,131 @@ static int event()
     return DURATION_IGNORE;
 }
 
+/***
+ * @brief: Convert the current TeamraceChannel value to the appropriate config value for comparison
+*/
+static uint8_t teamraceChannelToConfigValue()
+{
+    // SWITCH3b is 1,2,3,4,5,6,x,Mid
+    //             0 1 2 3 4 5    7
+    // Config values are Disabled,1,2,3,Mid,4,5,6
+    //                      0     1 2 3  4  5 6 7
+    uint8_t retVal = CRSF_to_SWITCH3b(ChannelData[config.GetTeamraceChannel()]);
+    switch (retVal)
+    {
+        case 0: // passthrough
+        case 1: // passthrough
+        case 2:
+            return retVal + 1;
+        case 3: // passthrough
+        case 4: // passthrough
+        case 5:
+            return retVal + 2;
+        case 7:
+            return 4; // "Mid"
+        default:
+            // CRSF_to_SWITCH3b should only return 0-5,7 but we must return a value
+            return 0;
+    }
```

Comment:
Would this be more easily done with an array lookup?

Reply:
Simple array lookups use more code, e.g. converting this code to
```cpp
    uint8_t retVal = CRSF_to_SWITCH3b(ChannelData[config.GetTeamraceChannel()]);
    constexpr uint8_t B3_TO_CONFIG[7] = { 1, 2, 3, 5, 6, 7, 4 };
    return B3_TO_CONFIG[retVal];
```
Is 24 bytes more code than the switch statement, and lacks the protection of returning 0 in case somehow `CRSF_to_SWITCH3b` returns something outside the bounds. Using a uint32_t for the const adds 48 bytes + 32 bytes RAM. Those quick LUTs are incredibly fat.

Also I think it is clearer why the numbers are derived instead of just `1, 2, 3, 5, 6, 7, 4`.

Reply:
Fat indeed!